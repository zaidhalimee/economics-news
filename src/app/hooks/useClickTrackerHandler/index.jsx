/* eslint-disable no-console */
import { useContext, useCallback, useState } from 'react';

import { EventTrackingContext } from '../../contexts/EventTrackingContext';
import useTrackingToggle from '../useTrackingToggle';
import OPTIMIZELY_CONFIG from '../../lib/config/optimizely';
import { sendEventBeacon } from '../../components/ATIAnalytics/beacon/index';
import { ServiceContext } from '../../contexts/ServiceContext';
import { isValidClick } from './clickTypes';

const EVENT_TYPE = 'click';

const useClickTrackerHandler = (props = {}) => {
  const preventNavigation = props?.preventNavigation;
  const componentName = props?.componentName;
  const url = props?.url;
  const advertiserID = props?.advertiserID;
  const format = props?.format;
  const optimizely = props?.optimizely;
  const optimizelyMetricNameOverride = props?.optimizelyMetricNameOverride;
  const detailedPlacement = props?.detailedPlacement;

  const { trackingIsEnabled } = useTrackingToggle(componentName);
  const [clicked, setClicked] = useState(false);
  const eventTrackingContext = useContext(EventTrackingContext);

  const { pageIdentifier, platform, producerId, statsDestination } =
    eventTrackingContext;
  const campaignID = props?.campaignID || eventTrackingContext?.campaignID;

  const { service } = useContext(ServiceContext);

  return useCallback(
    async event => {
      const shouldRegisterClick = [
        trackingIsEnabled,
        !clicked,
        isValidClick(event),
      ].every(Boolean);
      if (shouldRegisterClick) {
        setClicked(true);

        const shouldSendEvent = [
          campaignID,
          componentName,
          pageIdentifier,
          platform,
          producerId,
          service,
          statsDestination,
        ].every(Boolean);
        if (shouldSendEvent) {
          const nextPageUrl = event?.currentTarget?.href;

          event.stopPropagation();
          event.preventDefault();

          if (optimizely) {
            const eventName = OPTIMIZELY_CONFIG.viewClickAttributeId;

            const overrideAttributes = {
              ...optimizely.user.attributes,
              [`clicked_${eventName}`]: true,
            };

            optimizely.track(
              optimizelyMetricNameOverride
                ? `${optimizelyMetricNameOverride}_clicks`
                : 'component_clicks',
              optimizely.user.id,
              overrideAttributes,
            );
          }

          const optimizelyVariation =
            optimizely?.getVariation(OPTIMIZELY_CONFIG.ruleKey) || null;

          try {
            await sendEventBeacon({
              type: EVENT_TYPE,
              campaignID,
              componentName,
              format,
              pageIdentifier,
              platform,
              producerId,
              service,
              advertiserID,
              statsDestination,
              url,
              detailedPlacement,
              ...(optimizelyVariation &&
                optimizelyVariation !== 'off' && {
                  experimentVariant: optimizelyVariation,
                }),
            });
          } finally {
            if (nextPageUrl && !preventNavigation) {
              if (optimizely) {
                optimizely.close();
              }
              window.location.assign(nextPageUrl);
            }
          }
        }
      }
    },
    [
      trackingIsEnabled,
      clicked,
      campaignID,
      componentName,
      pageIdentifier,
      platform,
      preventNavigation,
      producerId,
      service,
      statsDestination,
      url,
      advertiserID,
      format,
      optimizely,
      optimizelyMetricNameOverride,
      detailedPlacement,
    ],
  );
};

export default useClickTrackerHandler;
