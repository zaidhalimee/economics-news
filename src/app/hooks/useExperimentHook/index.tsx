import useOperaMiniDetection from '#app/hooks/useOperaMiniDetection';
import { Services } from '#app/models/types/global';
import { useContext, useEffect, useState } from 'react';
import { ServiceContext } from '../../contexts/ServiceContext';

// Disabled due to bug in ts lint
// eslint-disable-next-line no-shadow
export enum Stages {
  STAGE_1 = 'stage_1',
  STAGE_2 = 'stage_2',
  STAGE_3 = 'stage_3',
  DEFAULT = 'default',
}

type ExperimentCriteria = Partial<{
  service: Services;
  isOperaMini: boolean;
  dataSaver: boolean;
  lowPower: boolean;
  noJs: boolean;
}>;

export type Navigator = {
  connection: { saveData: boolean };
  getBattery?: () => Promise<{ level: number }>;
};

const LOW_POWER_THRESHOLD = 0.2;

const determineStage = ({
  service,
  isOperaMini,
  dataSaver,
  lowPower,
  noJs,
}: ExperimentCriteria) => {
  if (noJs) {
    return Stages.STAGE_1;
  }

  if (service !== 'mundo' && !lowPower && !dataSaver && !isOperaMini) {
    return Stages.STAGE_3;
  }

  if (service === 'mundo' || dataSaver || isOperaMini || lowPower) {
    return Stages.STAGE_2;
  }

  return Stages.DEFAULT;
};

const useExperimentHook = () => {
  const [lowPower, setLowPower] = useState(false);
  const [dataSaver, setSaveDataMode] = useState(false);
  const [noJs, setNoJs] = useState(true);
  const isOperaMini = useOperaMiniDetection();
  const { service } = useContext(ServiceContext);

  useEffect(() => {
    const initialiseDeviceStates = async () => {
      const nav = navigator as unknown as Navigator;
      const saveDataMode = nav.connection?.saveData;
      if (nav.getBattery) {
        const manager = await nav.getBattery();
        const { level } = manager;
        const isLowPower = level <= LOW_POWER_THRESHOLD;
        setLowPower(isLowPower);
      }
      setSaveDataMode(saveDataMode);
    };
    initialiseDeviceStates();
    setNoJs(false);
  }, []);

  const stage = determineStage({
    isOperaMini,
    service,
    dataSaver,
    lowPower,
    noJs,
  });

  return stage;
};

export default useExperimentHook;
