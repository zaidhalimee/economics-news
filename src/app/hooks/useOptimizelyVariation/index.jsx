/* eslint-disable react-hooks/rules-of-hooks */
import { useState, useEffect } from 'react';
import { useDecision } from '@optimizely/react-sdk';

const isClientSide = true;

// ALTHOUGH THIS FUNCTION BREAKS REACT RULES BY USING CONDITIONAL HOOKS,
// WE CAN SAFELY DO SO SINCE isClientSide IS A CONSTANT AND THEREFORE GUARANTEES THAT
// EACH HOOK WILL BE CALLED IN THE EXACT SAME ORDER UPON INITAL RENDER.
const useOptimizelyVariation = (
  flagId,
  overrideAttributes = {},
  useClientSide = isClientSide,
) => {
  if (useClientSide) {
    const [decision, isClientReady, didTimeout] = useDecision(
      flagId,
      {
        autoUpdate: true,
      },
      { overrideAttributes },
    );

    const [variation, setVariation] = useState(null);

    useEffect(() => {
      if (isClientReady && !didTimeout) {
        setVariation(decision.variationKey);

        // Exposes the variation key to the window object for use in the global scope
        // Could potentially be moved to a Context and accessed that way
        window.optimizelyVariation = decision.variationKey;
      }
    }, [isClientReady, decision.variationKey, didTimeout]);

    return variation;
  }

  return true;
};

export default useOptimizelyVariation;
