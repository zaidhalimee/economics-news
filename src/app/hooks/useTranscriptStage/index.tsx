import useOperaMiniDetection from '#app/hooks/useOperaMiniDetection';
import { Services } from '#app/models/types/global';
import { useContext, useEffect, useState } from 'react';
import { ServiceContext } from '../../contexts/ServiceContext';

// Disabled due to bug in ts lint
// eslint-disable-next-line no-shadow
export enum Stages {
  STAGE_1 = `No js stage`,
  STAGE_2 = `Qualifies for transcript`,
  STAGE_3 = `Does not qualify for transcript`,
}

type ExperimentCriteria = Partial<{
  service: Services;
  isOperaMini: boolean;
  dataSaver: boolean;
  lowPower: boolean;
  noJs: boolean;
  hasTranscript: boolean;
}>;

export type Navigator = {
  connection: { saveData: boolean };
  getBattery?: () => Promise<{ level: number }>;
};

const LOW_POWER_THRESHOLD = 0.2;

const determineStage = ({
  service,
  isOperaMini,
  dataSaver,
  lowPower,
  noJs,
  hasTranscript,
}: ExperimentCriteria) => {
  if (noJs) {
    return Stages.STAGE_1;
  }

  if (
    (service === 'mundo' || dataSaver || isOperaMini || lowPower) &&
    hasTranscript
  ) {
    return Stages.STAGE_2;
  }

  return Stages.STAGE_3;
};

const useTranscriptStage = (hasTranscript: boolean) => {
  const [lowPower, setLowPower] = useState(false);
  const [dataSaver, setSaveDataMode] = useState(false);
  const [noJs, setNoJs] = useState(true);
  const isOperaMini = useOperaMiniDetection();
  const { service } = useContext(ServiceContext);

  useEffect(() => {
    const initialiseDeviceStates = async () => {
      const nav = navigator as unknown as Navigator;
      const saveDataMode = nav.connection?.saveData;
      if (nav.getBattery) {
        const manager = await nav.getBattery();
        const { level } = manager;
        const isLowPower = level <= LOW_POWER_THRESHOLD;
        setLowPower(isLowPower);
      }
      setSaveDataMode(saveDataMode);
    };
    initialiseDeviceStates();
    setNoJs(false);
  }, []);

  const stage = determineStage({
    isOperaMini,
    service,
    dataSaver,
    lowPower,
    noJs,
    hasTranscript,
  });

  return stage;
};

export default useTranscriptStage;
