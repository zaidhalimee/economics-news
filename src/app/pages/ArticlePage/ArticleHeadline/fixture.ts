const fixture = {
  type: 'headline',
  blocks: [
    {
      id: 'b209a5ab',
      type: 'text',
      model: {
        blocks: [
          {
            id: '1fe1e342',
            type: 'paragraph',
            model: {
              text: "'Inama ya gicuti' hagati ya Tshisekedi na Kagame - Qatar yabishoboye ite?",
              blocks: [
                {
                  id: 'bc738584',
                  type: 'fragment',
                  model: {
                    text: "'Inama ya gicuti' hagati ya Tshisekedi na Kagame - Qatar yabishoboye ite?",
                    attributes: [],
                  },
                  position: [1, 1, 1, 1],
                },
              ],
            },
            position: [1, 1, 1],
          },
        ],
      },
      position: [1, 1],
    },
  ],
};

export default fixture;
