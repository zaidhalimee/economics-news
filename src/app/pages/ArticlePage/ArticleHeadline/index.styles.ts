import { css, Theme } from '@emotion/react';
import pixelsToRem from '#app/utilities/pixelsToRem';

export default {
  liteCTAContainer: ({ mq }: Theme) =>
    css({
      marginBottom: `${pixelsToRem(20)}rem`,
      [mq.GROUP_3_MIN_WIDTH]: {
        marginBottom: `${pixelsToRem(28)}rem`,
      },
    }),
  loadingContainer: () =>
    css({
      height: `${pixelsToRem(44)}rem`,
    }),
  liteCTA: ({ spacings, mq }: Theme) =>
    css({
      display: 'inline-block',
      marginInlineStart: `${spacings.FULL}rem`,

      [mq.GROUP_2_MIN_WIDTH]: {
        marginInlineStart: `${spacings.DOUBLE}rem`,
      },
      [mq.GROUP_4_MIN_WIDTH]: {
        marginInlineStart: 0,
      },
    }),
  headlineStylesOverride: ({ spacings, mq }: Theme) =>
    css({
      paddingBottom: `${spacings.HALF}rem`,
      [mq.GROUP_3_MIN_WIDTH]: {
        paddingBottom: `${pixelsToRem(12)}rem`,
      },
    }),
  displayNone: () =>
    css({
      display: 'none',
    }),
};
