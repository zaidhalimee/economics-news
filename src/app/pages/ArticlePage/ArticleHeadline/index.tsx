/** @jsx jsx */
/* @jsxFrag React.Fragment */
import { jsx } from '@emotion/react';
import React, { useContext } from 'react';
import { RequestContext } from '#contexts/RequestContext';
import useViewTracker from '#app/hooks/useViewTracker';
import useToggle from '#hooks/useToggle';
import CallToActionLinkWithChevron from '#app/components/CallToActionLinkWithChevron';
import { ServiceContext } from '#contexts/ServiceContext';
import Headings from '#containers/Headings';
import useOptimizelyVariation from '#app/hooks/useOptimizelyVariation';
import OPTIMIZELY_CONFIG from '#lib/config/optimizely';
import OptimizelyPageViewTracking from '#app/legacy/containers/OptimizelyPageViewTracking';
import { OptimizelyContext } from '@optimizely/react-sdk';
import { ComponentToRenderProps } from '../types';
import styles from './index.styles';

const ArticleHeadline = (props: ComponentToRenderProps) => {
  const { pathname, isLite } = useContext(RequestContext);
  const { translations } = useContext(ServiceContext);
  const { optimizely } = useContext(OptimizelyContext);
  const eventTrackingData = { componentName: 'canonical-lite-cta', optimizely };
  const { enabled: showCTA } = useToggle('liteSiteCTA');
  const viewRef = useViewTracker(eventTrackingData);
  const titleVariation = useOptimizelyVariation(
    OPTIMIZELY_CONFIG.flagKey,
    true,
  );

  let articleDataSavingLinkText =
    translations?.liteSite?.articleDataSavingLinkText ?? 'Data saving version';

  const titleExperimentVariations = translations.liteSite?.experiment;

  if (titleExperimentVariations && titleVariation != null) {
    articleDataSavingLinkText =
      titleExperimentVariations[titleVariation as unknown as string] ??
      articleDataSavingLinkText;
  }

  const showLiteCTAOnCanonical: boolean = !isLite && showCTA;

  return (
    <>
      <Headings
        className="article-heading"
        {...props}
        {...(showLiteCTAOnCanonical && {
          css: styles.headlineStylesOverride,
        })}
      />
      {showLiteCTAOnCanonical && (
        <>
          <div
            css={[
              styles.loadingContainer,
              styles.liteCTAContainer,
              titleVariation && styles.displayNone,
            ]}
            data-e2e="to-lite-site-loading"
          />
          <div
            css={[
              styles.liteCTAContainer,
              !titleVariation && styles.displayNone,
            ]}
            ref={viewRef}
            data-e2e="to-lite-site"
          >
            <CallToActionLinkWithChevron
              eventTrackingData={eventTrackingData}
              href={`${pathname}.lite`}
              css={styles.liteCTA}
            >
              {articleDataSavingLinkText}
            </CallToActionLinkWithChevron>
            <OptimizelyPageViewTracking />
          </div>
        </>
      )}
    </>
  );
};

export default ArticleHeadline;
