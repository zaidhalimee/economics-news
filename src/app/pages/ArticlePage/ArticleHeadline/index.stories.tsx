import React from 'react';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { ToggleContextProvider } from '#contexts/ToggleContext';
import { RequestContextProvider } from '#contexts/RequestContext';
import { Services } from '#app/models/types/global';
import { ARTICLE_PAGE } from '../../../routes/utils/pageTypes';
import ArticleHeadline from '.';
import fixture from './fixture';

type Props = {
  service: Services;
  isLite?: boolean;
  toggleEnabled?: boolean;
};

const ArticleHeadlineComponent = ({
  service,
  isLite,
  toggleEnabled = false,
}: Props) => (
  <ToggleContextProvider
    toggles={{
      liteSiteCTA: {
        enabled: toggleEnabled,
      },
    }}
  >
    <RequestContextProvider
      isLite={isLite}
      pageType={ARTICLE_PAGE}
      pathname="/pathname"
      service={service}
    >
      <ServiceContextProvider service={service}>
        <ArticleHeadline {...fixture} />
      </ServiceContextProvider>
    </RequestContextProvider>
  </ToggleContextProvider>
);

export default {
  title: 'components/ArticleHeadline',
  component: ArticleHeadline,
};

export const ArticleHeadlineWithLiteCTA = () => (
  <ArticleHeadlineComponent service="gahuza" toggleEnabled />
);

export const ArticleHeadlineWithLiteCTARightToLeft = () => (
  <ArticleHeadlineComponent service="arabic" toggleEnabled />
);

export const ArticleHeadlineDefault = () => (
  <ArticleHeadlineComponent service="news" />
);
