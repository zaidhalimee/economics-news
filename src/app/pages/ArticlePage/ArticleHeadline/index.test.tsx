import React from 'react';
import {
  act,
  render,
} from '#app/components/react-testing-library-with-providers';
import * as useOptimizelyVariation from '#app/hooks/useOptimizelyVariation';
import ArticleHeadline from '.';

jest.mock('#app/legacy/containers/OptimizelyPageViewTracking');

const headlineBlock = {
  id: 'c7298038',
  type: 'text',
  blocks: [
    {
      id: 'eed627ab',
      type: 'paragraph',
      model: {
        text: '‘Nziyamamariza kuba perezida w’Uburusiya Putin navaho’ - Umupfakazi wa Navalny',
        blocks: [
          {
            id: 'af816737',
            type: 'fragment',
            model: {
              text: '‘Nziyamamariza kuba perezida w’Uburusiya Putin navaho’ - Umupfakazi wa Navalny',
              attributes: [],
            },
          },
        ],
      },
    },
  ],
};

describe('ArticleHeadline - Lite Site CTA', () => {
  const useDecisionSpy = jest.spyOn(useOptimizelyVariation, 'default');

  afterEach(() => {
    jest.clearAllMocks();
  });

  it.each([
    { variation: 'control_text_only', expected: 'Inyandiko gusa' },
    {
      variation: 'variation_a_explore_data_friendly_version',
      expected: 'Koresha uburyo butwara amahera make',
    },
    {
      variation: 'variation_b_data_saving_version',
      expected: 'Uburyo buziganya amahera',
    },
    {
      variation: 'variation_c_read_data_saving_version',
      expected: 'Soma mu buryo buziganya amahera',
    },
    { variation: 'variation_d_lite_site', expected: 'Site yoroheje' },
    {
      variation: 'variation_e_2g_optimised_version',
      expected: '2G Ukoresheje uburyo busanzwe',
    },
    {
      variation: 'variation_f_low_data_version',
      expected: 'Uburyo butwara amahera make',
    },
    {
      variation: 'off',
      expected: 'Inyandiko gusa',
    },
  ])(
    'Should display $expected when the variation is $variation',
    async ({ variation, expected }) => {
      useDecisionSpy.mockReturnValueOnce(variation as unknown as true);

      let container;

      await act(async () => {
        ({ container } = await act(() => {
          return render(<ArticleHeadline {...headlineBlock} />, {
            service: 'gahuza',
            toggles: { liteSiteCTA: { enabled: true } },
          });
        }));
      });

      const titleSpan = (container as unknown as HTMLElement).querySelector(
        'div[data-e2e="to-lite-site"] div span span',
      );

      expect(titleSpan?.innerHTML).toBe(expected);
    },
  );

  it('Before hydration snapshot', async () => {
    useDecisionSpy.mockReturnValueOnce(null as unknown as true);

    let container;

    await act(async () => {
      ({ container } = await act(() => {
        return render(<ArticleHeadline {...headlineBlock} />, {
          service: 'gahuza',
          toggles: { liteSiteCTA: { enabled: true } },
        });
      }));
    });

    expect(container).toMatchSnapshot();
  });

  it('After hydration snapshot', async () => {
    useDecisionSpy.mockReturnValueOnce('off' as unknown as true);

    let container;

    await act(async () => {
      ({ container } = await act(() => {
        return render(<ArticleHeadline {...headlineBlock} />, {
          service: 'gahuza',
          toggles: { liteSiteCTA: { enabled: true } },
        });
      }));
    });

    expect(container).toMatchSnapshot();
  });
});
