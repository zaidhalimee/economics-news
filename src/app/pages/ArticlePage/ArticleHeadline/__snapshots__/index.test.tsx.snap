// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleHeadline - Lite Site CTA After hydration snapshot 1`] = `
.emotion-0 {
  margin-bottom: 1.25rem;
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    margin-bottom: 1.75rem;
  }
}

.emotion-1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  color: #141414;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0.75rem 0 0.75rem;
  display: inline-block;
  -webkit-margin-start: 0.5rem;
  margin-inline-start: 0.5rem;
}

.emotion-1:hover,
.emotion-1:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-1:focus,
.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (min-width: 25rem) {
  .emotion-1 {
    -webkit-margin-start: 1rem;
    margin-inline-start: 1rem;
  }
}

@media (min-width: 63rem) {
  .emotion-1 {
    -webkit-margin-start: 0;
    margin-inline-start: 0;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-3 {
  color: #141414;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: Helmet,Freesans,Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 700;
  vertical-align: middle;
  color: inherit;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-3 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-3 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

.emotion-4 {
  color: #141414;
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-family: Helmet,Freesans,Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 700;
  color: inherit;
  border-bottom: 0.0625rem solid #141414;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-4 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-4 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

a:visited .emotion-4 {
  color: #6E6E73;
  border-bottom: 0.0625rem solid #6E6E73;
}

a:focus .emotion-4,
a:hover .emotion-4 {
  border-bottom: 0.125rem solid #B80000;
  color: #B80000;
}

.emotion-5 {
  color: #141414;
  fill: currentColor;
  -webkit-margin-start: 0.5rem;
  margin-inline-start: 0.5rem;
  vertical-align: middle;
  width: 0.875rem;
  height: 0.875rem;
}

a:visited .emotion-5 {
  color: #6E6E73;
}

a:focus .emotion-5,
a:hover .emotion-5 {
  color: #B80000;
}

@media screen and (forced-colors: active) {
  .emotion-5 {
    fill: linkText;
  }

  a:visited .emotion-5 {
    fill: visitedText;
  }
}

<div>
  <div
    class="emotion-0"
    data-e2e="to-lite-site"
  >
    <a
      class="emotion-1"
      href="/news/articles/c0g992jmmkko.lite"
    >
      <div
        class="emotion-2"
      >
        <span
          class="emotion-3"
        >
          <span
            class="emotion-4"
          >
            Inyandiko gusa
          </span>
          <svg
            aria-hidden="true"
            class="emotion-5"
            focusable="false"
            height="12"
            viewBox="0 0 32 32"
            width="12"
          >
            <path
              d="M21.6 14.3L5.5 31h6.4l14.6-15L11.9 1H5.5l16.1 16.7v-3.4z"
            />
          </svg>
        </span>
      </div>
    </a>
  </div>
</div>
`;

exports[`ArticleHeadline - Lite Site CTA Before hydration snapshot 1`] = `
.emotion-0 {
  margin-bottom: 1.25rem;
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    margin-bottom: 1.75rem;
  }
}

.emotion-1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  color: #141414;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0.75rem 0 0.75rem;
  display: inline-block;
  -webkit-margin-start: 0.5rem;
  margin-inline-start: 0.5rem;
}

.emotion-1:hover,
.emotion-1:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-1:focus,
.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (min-width: 25rem) {
  .emotion-1 {
    -webkit-margin-start: 1rem;
    margin-inline-start: 1rem;
  }
}

@media (min-width: 63rem) {
  .emotion-1 {
    -webkit-margin-start: 0;
    margin-inline-start: 0;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-3 {
  color: #141414;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: Helmet,Freesans,Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 700;
  vertical-align: middle;
  color: inherit;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-3 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-3 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

.emotion-4 {
  color: #141414;
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-family: Helmet,Freesans,Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 700;
  color: inherit;
  border-bottom: 0.0625rem solid #141414;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-4 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-4 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

a:visited .emotion-4 {
  color: #6E6E73;
  border-bottom: 0.0625rem solid #6E6E73;
}

a:focus .emotion-4,
a:hover .emotion-4 {
  border-bottom: 0.125rem solid #B80000;
  color: #B80000;
}

.emotion-5 {
  color: #141414;
  fill: currentColor;
  -webkit-margin-start: 0.5rem;
  margin-inline-start: 0.5rem;
  vertical-align: middle;
  width: 0.875rem;
  height: 0.875rem;
}

a:visited .emotion-5 {
  color: #6E6E73;
}

a:focus .emotion-5,
a:hover .emotion-5 {
  color: #B80000;
}

@media screen and (forced-colors: active) {
  .emotion-5 {
    fill: linkText;
  }

  a:visited .emotion-5 {
    fill: visitedText;
  }
}

<div>
  <div
    class="emotion-0"
    data-e2e="to-lite-site"
  >
    <a
      class="emotion-1"
      href="/news/articles/c0g992jmmkko.lite"
    >
      <div
        class="emotion-2"
      >
        <span
          class="emotion-3"
        >
          <span
            class="emotion-4"
          >
            Inyandiko gusa
          </span>
          <svg
            aria-hidden="true"
            class="emotion-5"
            focusable="false"
            height="12"
            viewBox="0 0 32 32"
            width="12"
          >
            <path
              d="M21.6 14.3L5.5 31h6.4l14.6-15L11.9 1H5.5l16.1 16.7v-3.4z"
            />
          </svg>
        </span>
      </div>
    </a>
  </div>
</div>
`;
